### PR TITLE
[Common] common functions about time-sync and element type

### PR DESCRIPTION
**Pull request by jaeyun-jung**
1. add common functions about time-sync option (mux/merge)
2. refactor function name and param to handle sync option
3. add functions to get tensor type string and element size

Signed-off-by: Jaeyun Jung <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_common.c [hunk 1]
```diff
@@ -1027,16 +1027,94 @@ gst_tensors_typefind_function (GstTypeFind * tf, gpointer pdata)
   }
 }
 
+static const gchar *gst_tensor_time_sync_mode_string[] = {
+  [SYNC_NOSYNC] = "nosync",
+  [SYNC_SLOWEST] = "slowest",
+  [SYNC_BASEPAD] = "basepad",
+  [SYNC_END] = "error"
+};
+
```
Comment (jijoongmoon) [inline gst/nnstreamer/tensor_common.c]: It is good to be here in common for future use of other plugins.

File: gst/nnstreamer/tensor_common.h [hunk 2]
```diff
@@ -111,7 +111,8 @@ typedef struct
  * For the concrete definition of headers, please look at the wiki page of nnstreamer:
  * https://github.com/nnsuite/nnstreamer/wiki/Design-External-Save-Format-for-other-tensor-and-other-tensors-Stream-for-TypeFind
  */
-extern void gst_tensors_typefind_function (GstTypeFind * tf, gpointer pdata);
+extern void
+gst_tensors_typefind_function (GstTypeFind * tf, gpointer pdata);
```
Comment (jijoongmoon) [inline gst/nnstreamer/tensor_common.h]: This should go other commit.

File: gst/nnstreamer/tensor_common.c [hunk 3]
```diff
@@ -480,7 +480,7 @@ gst_tensors_info_get_types_string (const GstTensorsInfo * info)
     GString *types = g_string_new (NULL);
 
     for (i = 0; i < info->num_tensors; i++) {
-      g_string_append (types, tensor_element_typename[info->info[i].type]);
+      g_string_append (types, gst_tensor_get_type_string (info->info[i].type));
```
Comment (jijoongmoon) [inline gst/nnstreamer/tensor_common.c]: LGTM

File: gst/nnstreamer/nnstreamer.c [hunk 4]
```diff
@@ -71,12 +71,44 @@ NNSTREAMER_PLUGIN (tensor_reposrc);
       return FALSE; \
   } while (0);
 
+#if 0
+/**
```
Comment (jijoongmoon) [inline gst/nnstreamer/nnstreamer.c]: I think more explanation is required for this move.

File: nnstreamer_example/custom_example_scaler/nnstreamer_customfilter_example_scaler.c [hunk 5]
```diff
@@ -140,7 +141,7 @@ pt_invoke (void *private_data, const GstTensorFilterProperties * prop,
       prop->output_meta.info[0].dimension[3]);
   assert (prop->input_meta.info[0].type == prop->output_meta.info[0].type);
 
-  elementsize = tensor_element_size[prop->input_meta.info[0].type];
+  elementsize = gst_tensor_get_element_size (prop->input_meta.info[0].type);
```
Comment (jijoongmoon) [inline nnstreamer_example/custom_example_scaler/nnstreamer_customfilter_example_scaler.c]: LGTM

File: gst/nnstreamer/nnstreamer.c (hunk 4 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer.c]: In the description of gst_type_find_register()
 * Registers a new typefind function to be used for typefinding. After
 * registering this function will be available for typefinding.
 * This function is typically called during an element's plugin initialization.

Currently nnstreamer does not register type-find function, so I blocked this.

File: gst/nnstreamer/tensor_common.h (hunk 2 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common.h]: I added new PR about type-find function.

File: gst/nnstreamer/nnstreamer.c (hunk 4 above)
Comment (jaeyun-jung) [reply gst/nnstreamer/nnstreamer.c]: I added new PR about type-find function. ( PR #1417 )

File: gst/nnstreamer/tensor_common.c [hunk 6]
```diff
@@ -1027,16 +1063,94 @@ gst_tensors_typefind_function (GstTypeFind * tf, gpointer pdata)
   }
 }
 
+static const gchar *gst_tensor_time_sync_mode_string[] = {
+  [SYNC_NOSYNC] = "nosync",
+  [SYNC_SLOWEST] = "slowest",
+  [SYNC_BASEPAD] = "basepad",
+  [SYNC_END] = "error"
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_common.c]: For proper behavior of ```find_key_strv ```, you need to terminate this array with NULL. Add NULL to the last element.

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common.c]: Updated to set null when SYNC_END. ```[SYNC_END] = NULL```

**Pull request merged**